Build failure: ananicy-cpp

### Steps To Reproduce

Steps to reproduce the behavior:
1. Update flake.lock and perform a nixos-rebuild.

### Build log

```

> [  4%] [clang] Building BPF object: ananicy_cpp
> clang: error: unsupported option '-fzero-call-used-regs=used-gpr' for target 'bpf'
> make[2]: *** [libananicycpp_bpf/CMakeFiles/ananicy_cpp_bpf_c.dir/build.make:78: libananicycpp_bpf/an…
> make[1]: *** [CMakeFiles/Makefile2:125: libananicycpp_bpf/CMakeFiles/ananicy_cpp_bpf_c.dir/all] Erro…
> make: *** [Makefile:136: all] Error 2

```

Full Log:
[ananicy-cpp-build-error.log](https://github.com/user-attachments/files/16147901/ananicy-cpp-build-error.log)

### Additional context
This is a clang build error related to the unsupported option '-fzero-call-used-regs=used-gpr'.

### Notify maintainers

<!--
@Artturin @JohnRTitor @diniamo
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.7-zen1, NixOS, 24.11 (Vicuna), 24.11.20240708.655a58a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - nixpkgs: `/nix/store/yzkx7gx2cn73dh3c7zg5rvgkp3yli1cd-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/an/ananicy-cpp/package.nix
+++ b/pkgs/by-name/an/ananicy-cpp/package.nix
@@ -55,7 +55,7 @@ clangStdenv.mkDerivation rec {
   ];
 
   # BPF A call to built-in function '__stack_chk_fail' is not supported.
-  hardeningDisable = [ "stackprotector" ];
+  hardeningDisable = [ "stackprotector" "zerocallusedregs" ];
 
   cmakeFlags = [
     "-DUSE_EXTERNAL_JSON=ON"

